internal/server/data/migrations.go

```diff
@@ -1054,6 +1055,59 @@ func makeIdxEmailsProvidersUnique() *migrator.Migration {
 	}
 }
 
+func addDestinationCredentials() *migrator.Migration {
+	return &migrator.Migration{
+		ID: "2022-11-25T14:46",
+		Migrate: func(tx migrator.DB) error {
+			// Postgres 12 doesn't support CREATE OR REPLACE for triggers.
+			_, err := tx.Exec(`
+				CREATE TABLE IF NOT EXISTS destination_credentials (
+					id 							 	bigint NOT NULL,
+					organization_id  	bigint NOT NULL,
+					expires_at 				timestamp with time zone,
+					update_index 			bigint NOT NULL,
+					user_id 					bigint NOT NULL,
+					destination_id 		bigint NOT NULL,
+					answered          bool NOT NULL DEFAULT false,
+					--
+					bearer_token text
+				);
+
+				CREATE INDEX IF NOT EXISTS idx_cred_req_org_dest on destination_credentials (organization_id, destination_id);
+
+				CREATE OR REPLACE FUNCTION destination_credential_insert_notify() RETURNS trigger
+					LANGUAGE PLPGSQL
+					AS $$
+				BEGIN
+					-- on insert, we notify connector listeners for this destination
+					PERFORM pg_notify(current_schema() || '.credreq_' || uidinttostr(NEW.organization_id) || '_' || uidinttostr(NEW.destination_id), NEW.id::TEXT);
+					RETURN NULL;
+				END; $$;
+
+				DROP TRIGGER IF EXISTS credreq_notify_insert_trigger ON destination_credentials;
+				CREATE TRIGGER credreq_notify_insert_trigger AFTER insert
+					ON destination_credentials
+					FOR EACH ROW EXECUTE FUNCTION destination_credential_insert_notify();
+
+				CREATE OR REPLACE FUNCTION destination_credential_update_notify() RETURNS trigger
+					LANGUAGE PLPGSQL
+					AS $$
+				BEGIN
+					-- on update, we notify user listeners for this specific id, waiting to login
+					PERFORM pg_notify(current_schema() || '.credreq_' || uidinttostr(NEW.organization_id) || '_' || uidinttostr(NEW.id), NEW.id::TEXT);
```

Comment:
Doesn't this channel need a different name? How is the code able to tell if the `.credreq_` prefix refers to a `<org_id>_<create_req_id>` (this notify) vs a `<org_id>_<dest_id>` (the above one).

I think we need a different channel name prefix. 

Since we renamed the type to DestinationCredentials, maybe `destcreds_id` vs `destcreds_dest` for the channel names?

Reply:
statistically these numbers are never going to collide, but different channel names makes sense too